ProjectLighthouse.Servers.API/Controllers/UserEndpoints.cs

```diff
@@ -50,6 +50,32 @@ public async Task<IActionResult> GetUser(string username)
         return this.Ok(user);
     }
 
+    /// <summary>
+    /// Searches for the user based on the query
+    /// </summary>
+    /// <param name="query">The search query</param>
+    /// <returns>A list of users</returns>
+    /// <response code="200">The list of users, if any were found</response>
+    /// <response code="404">No users matched the query</response>
+    [HttpGet("user_search")]
+    [ProducesResponseType(typeof(User), StatusCodes.Status200OK)]
```

Comment:
```suggestion
    [ProducesResponseType(typeof(User[]), StatusCodes.Status200OK)]
```
I don't know if this is required since this is mainly for the swagger self-documenting thing that doesn't even seem to really work... but probably still a good idea to look into because you're returning a list of users